File: 20220907/모범답안/20220907_06.js/postfix.js
function Priority(op)
{
  switch(op)
  {
    case "+":
    case "-":
      return 1;
    case "*":
    case "/":
      return 2;
  }
}

function solution(S)
{
  var result = "";
  var stack = [];
  var op = "";
  var temp = "";

  S = S.split(" ");

  for (var i = 0; i < S.length; i++)
  {
    op = S[i];

    switch(op)
    {
      case "(":
        stack.push(op);
        break;
      case ")":
        while(stack.length > 0)
        {
          temp = stack.pop();

          if(temp == "(")
          {
            break;
          }

          result += (" " + temp);
        }
        break;
      case "+":
      case "-":
      case "*":
      case "/":
        while(stack.length != 0 && Priority(op) <= Priority(stack[stack.length - 1]))
        {
          temp = stack.pop();
          result += (" " + temp);
        }

        stack.push(op);
        break;
      default:
        if(result == "")
        {
          result += op;
        }
        else{
          result += (" " + op);
        }

        break;
    }
  }

  while(stack.length > 0)
  {
    result += (" " + stack.pop());
  }


  return result
}

File: 20220907/모범답안/20220907_06.js/postfix.py
def priority(op):
    if op == "+" or op == "-":
        return 1
    elif op == "*" or op == "/":
        return 2

def solution(S):
    result =""
    stack = []
    op = ""

    S = S.split(" ")

    for s in S:
        if s == "(":
            stack.append(s)
        elif s == ")":
            while len(stack) > 0:
                temp = stack.pop()
                if temp == "(": break
                result += (" " + temp)
        elif s == "+" or s == "-" or s == "*" or s == "/":
            while len(stack) != 0 and priority(s) <= priority(stack[len(stack)-1]):
                result += (" " + stack.pop())

            stack.append(s)
        else:
            result += (" " + s)

    while len(stack) > 0:
        result += (" " + stack.pop())

    return result.strip()
